### PR TITLE
[DOCS][MINOR] Scaladoc fixes (aka typo hunting)

### PR DESCRIPTION
**Pull request by jaceklaskowski**
## What changes were proposed in this pull request?

Minor changes to scaladoc

## How was this patch tested?

Local build

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 1]
```diff
@@ -72,24 +72,24 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   }
 
   /**
-   * Return all metadata that describes more details of this SparkPlan.
+   * Returns all metadata that describes more details of this SparkPlan.
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala]: You can even turn this into `@return` but no big deal.

File: sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala [hunk 2]
```diff
@@ -35,12 +35,13 @@ import org.apache.spark.sql.types.NumericType
 import org.apache.spark.sql.types.StructType
 
 /**
- * A set of methods for aggregations on a `DataFrame`, created by `Dataset.groupBy`.
+ * A set of methods for aggregations on a `DataFrame`, created by [[Dataset#groupBy groupBy]],
+ * [[Dataset#cube cube]] or [[Dataset#rollup rollup]] (and also [[pivot]]).
  *
- * The main method is the agg function, which has multiple variants. This class also contains
- * convenience some first order statistics such as mean, sum for convenience.
+ * The main method is the [[agg]] function, which has multiple variants. This class also contains
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala]: These look fine but have you had a chance to run the doc generation, to make sure scaladoc is OK with it?

**Review by srowen — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala (hunk 2 above)
Comment (jaceklaskowski) [inline sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala]: No scaladoc. I was focused on javadoc and seems I missed it. Fixed.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala [hunk 3]
```diff
@@ -153,19 +153,24 @@ case class WindowExec(
...
     type FrameKey = (String, FrameType, Option[Int], Option[Int])
     type ExpressionBuffer = mutable.Buffer[Expression]
-    val framedFunctions = mutable.Map.empty[FrameKey, (ExpressionBuffer, ExpressionBuffer)]
+    type WindowExpressionBuffer = mutable.Buffer[WindowExpression]
+    val framedFunctions = mutable.Map.empty[FrameKey, (WindowExpressionBuffer, ExpressionBuffer)]
 
     // Add a function and its function to the map for a given frame.
-    def collect(tpe: String, fr: SpecifiedWindowFrame, e: Expression, fn: Expression): Unit = {
+    def collect(
+      tpe: String, fr: SpecifiedWindowFrame, e: WindowExpression, fn: Expression): Unit =
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala]: This changed a type, which could be fine as it's an internal method, but why should it be the more specific type? this isn't a doc change

Comment (jaceklaskowski) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala]: I'll revert it to not "pollute" the PR. It was much easier to read the code when the internals were less generic (that does nothing at all).

**Comment by jaceklaskowski** [comment]
Hey @srowen could you review the changes again and accept possibly? Thanks!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala [hunk 4]
```diff
@@ -153,12 +153,13 @@ case class WindowExec(
...
   /**
-   * Collection containing an entry for each window frame to process. Each entry contains a frames'
-   * WindowExpressions and factory function for the WindowFrameFunction.
+   * Collection containing an entry for each window frame to process. Each entry contains a frame's
+   * [[WindowExpression]]s and factory function for the WindowFrameFunction.
    */
   private[this] lazy val windowFrameExpressionFactoryPairs = {
     type FrameKey = (String, FrameType, Option[Int], Option[Int])
     type ExpressionBuffer = mutable.Buffer[Expression]
+    type WindowExpressionBuffer = mutable.Buffer[WindowExpression]
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala]: @jaceklaskowski this is still not reverted. Please, back out any non-doc changes. I can't spend 4 rounds of review on things like this, it's not fair to anyone else waiting

Comment (jaceklaskowski) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala]: Sorry. Reverted the change manually and looks like it slipped through the cracks. Apologies.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**